pkg/injection/codemodule/installer/image/unpack.go

```diff
@@ -44,7 +49,14 @@ func (installer Installer) pullImageInfo(imageName string) (*containerv1.Image,
 		return nil, errors.WithMessagef(err, "parsing reference %q:", imageName)
 	}
 
-	image, err := remote.Image(ref, remote.WithContext(context.TODO()), remote.WithAuthFromKeychain(installer.keychain), remote.WithTransport(installer.transport))
+	image, err := remote.Image(ref, remote.WithContext(context.TODO()),
+		remote.WithAuthFromKeychain(installer.keychain),
+		remote.WithTransport(installer.transport),
+		remote.WithPlatform(containerv1.Platform{
+			OS:           defaultOS,
```

Comment:
do you have to pass OS?

Reply:
pretty sure yes
the "true" default value would be an empty string, which is probably problematic if you look at an image manifest, because you tend to see stuff like:
![image](https://github.com/Dynatrace/dynatrace-operator/assets/31651557/addfef8c-2411-4180-a4e1-6856fe576af8)

Reply:
yes, because in docker it's called platform=${platform-os}/${platform-arch},
in podman cli client it's has separate options to combine.

Reply:
@0sewa0 is any chance to test it?

Reply:
test to not specify?

Reply:
yeah, but it doesn't make any sense. Ignore my comment

Reply:
🫡